src/libcore/time.rs

```diff
@@ -30,6 +30,7 @@ const NANOS_PER_MILLI: u32 = 1_000_000;
 const NANOS_PER_MICRO: u32 = 1_000;
 const MILLIS_PER_SEC: u64 = 1_000;
 const MICROS_PER_SEC: u64 = 1_000_000;
+const MAX_NANOS_F64: f64 = ((u64::MAX as u128 + 1)*(NANOS_PER_SEC as u128) - 1) as f64;
```

Comment:
The `- 1` here doesn't change the value because `f64` doesn't have enough precision so this constant ends up being exactly 1ns greater than the largest possible `Duration`. Either this constant needs to be changed to the next smallest `f64` or the check below needs to be changed to `nanos >= MAX_NANOS_F64`. `from_float_secs(2f64.powi(64))` should trigger an overflow but currently it returns a `Duration` of 0ns.

Reply:
Ah, yes, it should've been `>=`. Fixed.